Sources/funnet/Core/Endpoint.swift

```diff
@@ -7,94 +7,32 @@
-    var timeout: TimeInterval { get set }
-    var postData: Data? { get set }
-    var dataStream: InputStream? { get set }
-}
-
-public struct Endpoint: EndpointProtocol {
+public struct Endpoint {
     public var httpMethod: String = "GET"
     public var httpHeaders: [String: String] = [:]
     public var path: String = ""
-    public var getParams: [String: Any] = [:]
+    public var getParams: [URLQueryItem] = []
     public var timeout: TimeInterval = 60
     public var postData: Data?
     public var dataStream: InputStream?
     
     public init() {}
-}
-
-public extension EndpointProtocol {
-    mutating func addHeaders(headers: [String: String]) {
+    
+    public mutating func addHeaders(headers: [String: String]) {
         for key in headers.keys {
             httpHeaders[key] = headers[key]
         }
     }
     
-    mutating func addModelData<E: Encodable>(model: E, encoder: JSONEncoder = JSONEncoder()) {
-        if let data = try? encoder.encode(model) {
-            self.postData = data
-        }
+    public mutating func addGetParams(params: [URLQueryItem]) {
+        getParams.append(contentsOf: params)
     }
-    
-    mutating func addGetParams(params: [String: String]) {
-        for key in params.keys {
-            getParams[key] = params[key]
-        }
-    }
-}
-
-public func dataSetter<M, T>(from model: M) -> (inout T) -> Void where M: Encodable, T: EndpointProtocol {
-    return { (endpoint: inout T) in
-        endpoint.addModelData(model: model)
-    }
-}
-
-public func addJsonHeaders<T>(_ endpoint: inout T) where T: EndpointProtocol {
-    endpoint.addHeaders(headers: ["Content-Type": "application/json", "Accept": "application/json"])
-}
-
-public func setToGet<T>(_ endpoint: inout T) where T: EndpointProtocol {
-    endpoint.httpMethod = "GET"
-}
-
-public func setToPost<T>(_ endpoint: inout T) where T: EndpointProtocol {
-    endpoint.httpMethod = "POST"
-}
-
-public func setToPut<T>(_ endpoint: inout T) where T: EndpointProtocol {
-    endpoint.httpMethod = "PUT"
-}
-
-public func setToPatch<T>(_ endpoint: inout T) where T: EndpointProtocol {
-    endpoint.httpMethod = "PATCH"
-}
-
-public func setToDelete<T>(_ endpoint: inout T) where T: EndpointProtocol {
-    endpoint.httpMethod = "DELETE"
-}
-
-public func setToGET<T>(_ endpoint: inout T) where T: EndpointProtocol {
-    setToGet(&endpoint)
-}
-
-public func setToPOST<T>(_ endpoint: inout T) where T: EndpointProtocol {
-    endpoint.httpMethod = "POST"
-}
-
-public func setToPUT<T>(_ endpoint: inout T) where T: EndpointProtocol {
-    endpoint.httpMethod = "PUT"
 }
 
-public func setToPATCH<T>(_ endpoint: inout T) where T: EndpointProtocol {
-    endpoint.httpMethod = "PATCH"
+public func addJsonHeaders(_ endpoint: inout Endpoint) {
+    endpoint.addHeaders(headers: jsonHeaders())
 }
 
-public func setToDELETE<T>(_ endpoint: inout T) where T: EndpointProtocol {
-    endpoint.httpMethod = "DELETE"
+public func jsonHeaders() -> [String: String] {
```

Comment:
Just curious, why does this need to be a function?

Reply:
Doesn't need to be a function, but it's nice to have it separate from `addJsonHeaders` so that people can use it directly on a URLRequest as well